tests/regression/roles/sonic_static_routes/defaults/main.yml

```diff
@@ -0,0 +1,191 @@
+---
+ansible_connection: httpapi
+module_name: static_routes
+
+vrf_1: VrfReg1
+vrf_2: VrfReg2
+
+preparations_tests:
+  init_vrf:
+    - 'ip vrf {{vrf_1}}'
+    - 'ip vrf {{vrf_2}}'
+tests:
+  - name: test_case_01
+    description: Add new static routes configuration
+    state: merged
+    input:
+      - vrf_name: 'default'
+        static_list:
+         - prefix: '2.0.0.0/8'
+           next_hops:
+             - index:
+                 interface: 'Ethernet4'
+               metric: 1
+               tag: 2
+               track: 3
+             - index:
+                next_hop: '3.0.0.0'
+               metric: 2
+               tag: 4
+               track: 8
+      - vrf_name: '{{vrf_1}}'
+        static_list:
+          - prefix: '3.0.0.0/8'
+            next_hops:
+              - index:
+                  blackhole: True
+                  interface: 'Ethernet4'
+                  network_instance: '{{vrf_2}}'
+                  next_hop: '5.0.0.0'
+                metric: 1
+                tag: 2
+                track: 3
+              - index:
+                  interface: 'eth0'
+                  network_instance: '{{vrf_2}}'
+                  next_hop: '4.0.0.0'
+                metric: 4
+                tag: 5
+                track: 6
+      - vrf_name: '{{vrf_2}}'
+        static_list:
+          - prefix: '1.0.0.0/8'
+            next_hops:
+              - index:
+                  interface: 'Ethernet6'
+                  network_instance: '{{vrf_1}}'
+              - index:
+                  interface: 'Ethernet8'
+                  next_hop: '2.0.0.0'
+          - prefix: '7.0.0.0/8'
+            next_hops:
+              - index:
+                  network_instance: '{{vrf_1}}'
+                  next_hop: '3.0.0.0'
+  - name: test_case_02
+    description: Modify static routes configuration
```

Comment:
It would be worthwhile to add a case that inserts a new next hop into an existing route.

The corresponding "delete" case, to verify deletion of a next hop by deleting the index for a next hop that still contains other attributes would also be worthwhile. This could be done by leaving at least one tag/track/metric attribute undeleted in test_case_3.